warehouse/router/upload.go

```diff
@@ -427,6 +431,37 @@ func (job *UploadJob) run() (err error) {
 	return nil
 }
 
+func (job *UploadJob) cleanupObjectStorageFiles() error {
+	destination := job.warehouse.Destination
+	cleanupObjectStorageFiles, _ := destination.Config["cleanupObjectStorageFiles"].(bool)
+	if cleanupObjectStorageFiles {
+		storageProvider := whutils.ObjectStorageType(destination.DestinationDefinition.Name, destination.Config, job.upload.UseRudderStorage)
+		fm, err := filemanager.New(&filemanager.Settings{Provider: storageProvider, Config: destination.Config})
+		if err != nil {
+			return err
+		}
```

Comment:
For config, we need to use `misc.GetObjectStorageConfig`
```suggestion
		fm, err := filemanager.New(&filemanager.Settings{
		Provider: storageProvider,
		Config: misc.GetObjectStorageConfig(misc.ObjectStorageOptsT{
			Provider:         storageProvider,
			Config:           destination.Config,
			UseRudderStorage: job.upload.UseRudderStorage,
			WorkspaceID:      job.upload.WorkspaceID,
		}),
	})
	if err != nil {
		return fmt.Errorf("creating file manager: %w", err)
	}
```

Reply:
Done. Perhaps we should refactor `filemanager.New` to abstract these details from the caller.